core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -277,6 +286,22 @@ static TableMetadata fromJson(FileIO io, InputFile file, JsonNode node) {
           schema, TableMetadata.INITIAL_SPEC_ID, node.get(PARTITION_SPEC)));
     }
 
+    JsonNode sortOrderArray = node.get(SORT_ORDERS);
+    List<SortOrder> sortOrders;
+    int defaultSortOrderId;
+    if (sortOrderArray != null) {
+      defaultSortOrderId = JsonUtil.getInt(DEFAULT_SORT_ORDER_ID, node);
+      ImmutableList.Builder<SortOrder> sortOrdersBuilder = ImmutableList.builder();
+      for (JsonNode sortOrder : sortOrderArray) {
+        sortOrdersBuilder.add(SortOrderParser.fromJson(schema, sortOrder));
+      }
+      sortOrders = sortOrdersBuilder.build();
+    } else {
+      SortOrder defaultSortOrder = SortOrder.unsorted();
```

Comment:
If the `SortOrder` list is required by v2, then I think that we should add a check here that the table is v1. That way, we fail to read a table and force recovery if a v2 table doesn't have an order list.

Reply:
I tried that but we need to adapt tests. So, I've created #1419 to address this in a follow-up.